File: 프로그래머스/분할정복/유사 칸토어 비트열.cpp
#include <string>
#include <vector>

using namespace std;

bool kant(long long i){ //64bit 정수형
    while (i > 0){
        if (i % 5 == 2){
            return false;
        }
        i /= 5;
    }
    return true;
}

int solution(int n, long long l, long long r) {
    int answer = 0;
    
    for (long long i = l - 1; i < r; i++){
        if (kant(i)){
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/분할정복/유사 칸토어 비트열.py
def solution(n, l, r):
    def kant(i):
        while(i > 0): 
            if i % 5 == 2: # 인덱스가 중앙이면 0
                return False
            i //= 5 # 인덱스가 중앙이 아니더라도 큰 묶음 기준 가운데일 수 있음
        return True
        
    answer = 0
        
    for i in range(l - 1, r):
        if kant(i):
            answer += 1
                
    return answer